docker/entrypoint.sh

```diff
@@ -1,24 +1,108 @@
 #!/bin/bash
+
+show_progress() {
+    message="$1"
+    current="$2"
+    total="$3"
+    percent="$4"
+
+    done=$(bc <<< "scale=0; 40 * ${percent%.*} / 100" )
+    todo=$(bc <<< "scale=0; 40 - $done" )
+    done_sub_bar=$(printf "%${done}s" | tr " " "#")
+    todo_sub_bar=$(printf "%${todo}s" | tr " " "-")
+    echo -e "$message $current/$total [${done_sub_bar}${todo_sub_bar}] ${percent}%"
+}
+
+node_verification() {
+     REGEX="^.?\[.*\].?\[.*\].?(Validating|Validated) chunk no\. ([0-9]+) out of ([0-9]+)\. Progress: ([0-9]+\.[0-9][0-9])\%.*$"
+     echo -e "Node verification..."
+     rm -f $CARDANO_NODE_SOCKET_PATH
+     while [ ! -S "$CARDANO_NODE_SOCKET_PATH" ] ; do
+          line=$(tail -n 1 /logs/node.log)
+          if [[ "$line" =~ $REGEX ]]; then
+               current="${BASH_REMATCH[2]}"
+               total="${BASH_REMATCH[3]}"
+               percent="${BASH_REMATCH[4]}"
+
+               show_progress "Node verification: Chunk" $current $total $percent
+          fi
+          sleep 1
+    done
+    echo -e "Node verification: DONE$(printf "%100s")"
+}
+
+node_synchronization() {
+    echo -e "Node synchronization..."
+    epoch_length=$(jq -r .epochLength $GENESIS_SHELLEY_PATH)
+    slot_length=$(jq -r .slotLength $GENESIS_SHELLEY_PATH)
+    byron_slot_length=$(( $(jq -r .blockVersionData.slotDuration $GENESIS_BYRON_PATH) / 1000 ))
+    byron_epoch_length=$(( $(jq -r .protocolConsts.k $GENESIS_BYRON_PATH) * 10 ))
+    byron_start=$(jq -r .startTime $GENESIS_BYRON_PATH)
+    byron_end=$((byron_start + $HARDFORK_EPOCH * byron_epoch_length * byron_slot_length))
+    byron_slots=$(($HARDFORK_EPOCH * byron_epoch_length))
+    now=$(date +'%s')
+    expected_slot=$((byron_slots + (now - byron_end) / slot_length))
+
+    sync_progress=0
+    while (( ${sync_progress%.*} < 100 )) ; do
+        current_status=$(cardano-cli query tip $NETWORK_STR)
+        current_slot=$(echo $current_status | jq -r '.slot')
+        sync_progress=$(echo $current_status | jq -r '.syncProgress')
+
+        show_progress "Node synchronization: Slot" $current_slot $expected_slot $sync_progress
+        sleep 1
+    done
+    echo -e "Node synchronization: DONE$(printf "%100s")"
+}
+
+database_initialisation() {
+    chown postgres:postgres /var/lib/postgresql/$PG_VERSION/main
+    chmod 0700 /var/lib/postgresql/$PG_VERSION/main
+    echo "postgres" >> /tmp/password
+    initdb_command="/usr/lib/postgresql/$PG_VERSION/bin/initdb --pgdata=/var/lib/postgresql/$PG_VERSION/main --auth=md5 --auth-local=md5 --auth-host=md5 --username=postgres --pwfile=/tmp/password"
+    sudo -H -u postgres bash -c "$initdb_command"
+}
+
+echo "entrypoint - network: $NETWORK"
+NETWORK_STR="--testnet-magic $PROTOCOL_MAGIC"
+HARDFORK_EPOCH=1
+if [[ "$NETWORK" == "mainnet" ]]; then
+    NETWORK_STR="--mainnet"
+    HARDFORK_EPOCH=208
```

Comment:
@Kammerlo I found these values on google, but couldn't find a description of what they are. Do you know where I can read about it and how I can check that it is correct.